TUTORIAL.md

```diff
@@ -0,0 +1,245 @@
+
+  <div class="field">
+    <label for="post_name">Name</label><br>
+    <input type="text" name="post[name]" id="post_name" /><br>
+    <label for="post_content">Content</label><br>
+    <textarea name="post[content]" id="post_content">
+    </textarea><br>
+    Tags:
+  </div>
+
+  <div class="field">
+    <label for="post_tags_attributes_0_name">Name</label>
+    <input type="text" name="post[tags_attributes][0][name]" id="post_tags_attributes_0_name" />
+  
+    <label for="post_tags_attributes_1_name">Name</label>
+    <input type="text" name="post[tags_attributes][1][name]" id="post_tags_attributes_1_name" />
+  
+    <label for="post_tags_attributes_2_name">Name</label>
+    <input type="text" name="post[tags_attributes][2][name]" id="post_tags_attributes_2_name" />
+  </div>
+
+  <div class="actions">
+    <input type="submit" name="commit" value="Create Post" />
+  </div>
+
+</form>
+```
+
+In the first section of our form, we can see our form helper generated `post[name]` and `post[content]`. This is great, we have access to our post name and content.
+
+#### `fields_for`
+Let's focus on our tag fields for a second, how did Rails know to create such a nicely nested structure?
+
+If we look at our form, we will see `<%= f.fields_for(:tags) do |tag_form| %>`. Our `fields_for` accepts a `:tags` object and by convention introspects into our `Post` model to see if it has a way to accept this new object. It is specifically looking for a macro called `accepts_nested_attributes_for :tags`. If it finds it, it will generate the nicely nested form that we see above.
+
+In order to understand `accepts_nested_attributes_for`, we need to take a look at our models associations and the methods that were made available to us.
+
+***note that we have added `accepts_nested_attributes_for :tags` to our `Post` model**
+
+##Step 4: Set up Post model
+### Models
+
+```ruby
+class Post < ActiveRecord::Base
+  has_many :post_tags
+  has_many :tags, :through => :post_tags
+
+  accepts_nested_attributes_for :tags
+end
+
+class PostTag < ActiveRecord::Base
+  belongs_to :post
+  belongs_to :tag
+end
+
+class Tag < ActiveRecord::Base
+  has_many :post_tags
+  has_many :posts, :through => :post_tags
+end
+```
+By creating these associations, Rails gives us the methods below.
+
+```ruby
+tag_ids()                    Post (Post::GeneratedAssociationMethods)
+tag_ids=(ids)                Post (Post::GeneratedAssociationMethods)
+tags(*args)                  Post (Post::GeneratedAssociationMethods)
+tags=(value)                 Post (Post::GeneratedAssociationMethods)
+tags_attributes=(attributes) Post (Post::GeneratedAssociationMethods)
+```
+What we are interested in right now is `tags_attributes=(attributes)`
+
+As mentioned before, our `fields_for` is looking for `accepts_nested_attributes_for :tags`. But what exactly does this macro give us? Let's define our own and see.
```

Comment:
nope. The Post.new is what requires `accepts_nested_attributes_for` stuff. 

At least I'm pretty sure.